crypto/rsa.c

```diff
@@ -145,13 +145,13 @@ int RSA_decrypt(const RSA_CTX *ctx, const uint8_t *in_data,
                             uint8_t *out_data, int out_len, int is_decryption)
 {
     const int byte_size = ctx->num_octets;
-    int i = 0, size;
+    int i = 0, size = -1;
     bigint *decrypted_bi, *dat_bi;
     uint8_t *block = (uint8_t *)malloc(byte_size);
     int pad_count = 0;
 
     if (out_len < byte_size)        /* check output has enough size */
-        return -1;
+       goto error;
```

Comment:
Just an idea: Move malloc on line 150 after this condition and put it right after line 154. This way if an initial check fails also the memory does not get fragmented unnecessarily.